lms/djangoapps/courseware/url_helpers.py

```diff
@@ -52,3 +53,46 @@ def get_redirect_url(course_key, usage_key, request=None):
         )
     redirect_url += "?{}".format(urlencode({'activate_block_id': six.text_type(final_target_id)}))
     return redirect_url
+
+
+def get_microfrontend_redirect_url(course_key, path=None):
```

Comment:
Who is the caller of this? Is `path` always derived from sanitized information that we have (i.e. real blocks), or does it sometimes get provided by the browser calling?

Reply:
It may be that I should just remove this method from the PR.  It _was_ being called by redirects which I removed from this PR, and will be again when we put those back in.  It's not currently in use, which now that I say it out loud, probably means I should just pull it for now. 😄 

The path is derived from `path_to_location` when it's used, so yes, I believe it's always sanitized.